lib/admin-page.php

```diff
@@ -480,7 +480,22 @@ function classicpress_check_can_migrate() {
 
 	// Check: Conflicting Plugins
 	$plugins = get_option( 'active_plugins' );
-	if ( isset( $parameters['plugins'] ) && $plugins !== array_diff( $plugins, $parameters['plugins'] ) ) {
+	$plugin_headers = array( 'Name' => 'Plugin Name', 'RequiresWP'  => 'Requires at least' );
+	$declared_incompatible_plugins = array();
+
+	// Start by checking if plugins have declared they require WordPress 5.0 or higher
+	foreach ( $plugins as $plugin ) {
+		$plugin_data = get_file_data( WP_CONTENT_DIR . '/plugins/' . $plugin, $plugin_headers );
+		if ( version_compare( $plugin_data['RequiresWP'], '5.0' ) >= 0 ) {
```

Comment:
Does this 'RequiresWP' always exist? And does the version compare work with an empty string?

Reply:
Yes, 'RequiresWP' always exists, if not present in the plugin header it will be an empty string.
Yes, `version_compare` works with empty strings - it returns -1 and therefore passes the test.

Reply:
Similar to https://github.com/ClassicPress/ClassicPress-Migration-Plugin/pull/76/files#r684700044, mostly copy/pasted from that comment:

As written, I think this shows the same message as when we have added a plugin to our hardcoded list of incompatible plugins in the API. I think this should be a separate check instead. This check could then indicate which of the possible situations is present:

- The plugin actually declares compatibility with WP 4.9+ or earlier (all good)
- The plugin declares compatibility with WP 5.0+ or later (bad)
- The plugin does not declare a compatibility header (unknown). Better to call this situation out explicitly than relying on a non-intuitive behavior of `version_compare` if this header is not present.

This way we can tell the user exactly what is being checked.

Adding more checks will mean that the migration plugin output is a bit long... I actually think this is fine. Migration is a complicated process and there is a lot that can go wrong.

Reply:
Also, it should use `plugins_url()`.

Reply:
This is a file path, not a URL. `plugins_url()` won't do what is needed.

`WP_PLUGIN_DIR` is probably a better choice than `WP_CONTENT_DIR . '/plugins'` since this will still work for the odd cases where plugins are not in `wp-content/plugins`. Even though [the WP documentation says](https://codex.wordpress.org/Determining_Plugin_and_Content_Directories#Constants) these constants "should not be used directly by plugins or themes", this article is written from the perspective of a plugin needing to get _its own_ file path, and I actually don't think there is a standard way to get the full path of _another plugin_ without using the same constants that core uses.

Reply:
Patched updated to use WP_PLUGIN_DIR